skills_ml/algorithms/job_sampler/sampler.py

```diff
@@ -0,0 +1,77 @@
+import numpy as np
+import heapq as hq
+import random
+
+def reservoir(it, k):
+    """
+    Reservoir sampling with Random Sort from a job posting iterator.
+
+    Randomly choosing a sample of k items from a streaming iterator. Using random sort to implement the algorithm.
+    Basically, it's assigning random number as keys to each item and maintain k items with minimum value for keys,
+    which equals to assigning a random number to each item as key and sort items using these keys and take top k items.
+
+    Args:
+        it (iterator): Job posting iterator to sample from
+        k (int): Sample size
+
+    Returns:
+        list: The result sample of k items.
+    """
+    it = iter(it)
+    result = []
+    for i, datum in enumerate(it):
+        if i < k:
+            result.append(datum)
+        else:
+            j = random.randint(0, i-1)
+            if j < k:
+                result[j] = datum
+    return result
+
+
+def reservoir_weighted(it, k, weights):
+    """
+    Weighted reservoir Sampling from job posting iterator.
+
+    Randomly choosing a sample of k items from a streaming iterator based on the weights.
+
+    Args:
+        it (iterator): Job posting iterator to sample from. The format should be (job_posting, label)
+        k (int): Sample size
+        weights (dict): weight
+    """
+    heap = []
+    hkey = lambda w: np.power(np.random.uniform(0.0, 1.0), 1.0 / w)
+    for i, datum in enumerate(it):
+        weight = weights[datum[1]]
```

Comment:
What expectation is this placing on the input of the function? It looks like it is expecting every label in the iterator to be present as a key in the weights dict. This expectation is something we should specify in the docstring.